### PR TITLE
Fix Resources

### PR DESCRIPTION
**Pull request by tydahlwave**
I realized that I broke the resources page when I was integrating Realm and Firebase. To fix it, I removed the Resource subclasses. I felt that this also makes the code simpler and was the quickest option to fix it.

I also updated the UIImage extensions to use the ImageManager class to manage resources better.

Might also be worth looking at the ImageManager class which isn't part of this commit just in case you have any comments on that.

### CODE REVIEW

File: CruCentralCoast/Extensions/UIImageExtensions.swift [hunk 1]
```diff
@@ -10,15 +10,9 @@ import UIKit
...
 extension UIImage {
     static func downloadedFrom(url: URL, contentMode mode: UIViewContentMode = .scaleAspectFit, completion: @escaping (UIImage?) -> Void) {
-        URLSession.shared.dataTask(with: url) { data, response, error in
-            guard
-                let httpURLResponse = response as? HTTPURLResponse, httpURLResponse.statusCode == 200,
-                let mimeType = response?.mimeType, mimeType.hasPrefix("image"),
-                let data = data, error == nil,
-                let image = UIImage(data: data)
-                else { completion(nil); return }
+        ImageManager.instance.fetch(url) { image in
```
Comment (cstock123) [inline CruCentralCoast/Extensions/UIImageExtensions.swift:13]: I did take a look at the Image manager class and see what is happening here. I like it and I also agree that it makes the code much much more simple. Great job!

**Review by cstock123 — COMMENTED** [review]
I don't have many comments. The code seems very straight forward and I think the changes simplified it greatly. I also reviewed the ImageManager class and like the layout.

Side note: I like the layout in the ImageManager class where you have a property called "instance" which just creates an instance of the class. It also makes the extensions available. I like the flow and learned some stuff from reviewing. Thanks!

**Comment by tydahlwave** [comment]
Thanks for reviewing! That is called the Singleton pattern, where only one instance of a class is allowed to exist and it can be accessed from anywhere in the codebase

File: CruCentralCoast/Extensions/UIImageViewExtensions.swift [hunk 2]
```diff
@@ -12,17 +12,9 @@ import UIKit
...
-                let mimeType = response?.mimeType, mimeType.hasPrefix("image"),
-                let data = data, error == nil,
-                let image = UIImage(data: data)
-                else { return }
-            DispatchQueue.main.async() {
-                self.image = image
-            }
-            }.resume()
+        ImageManager.instance.fetch(url) { [weak self] image in
+            self?.image = image
```
Comment (tydahlwave) [inline CruCentralCoast/Extensions/UIImageViewExtensions.swift]: Just realized I forgot to make sure this occurs on the main thread

Comment (tydahlwave) [reply CruCentralCoast/Extensions/UIImageViewExtensions.swift]: Fixed

**Review by cstock123 — APPROVED** [review]

**Pull request merged**